internal/org/repository/eventsourcing/org_test.go

```diff
@@ -847,3 +848,561 @@ func TestOrgDomainRemovedAggregates(t *testing.T) {
+
+			if !tt.res.wantErr && len(agg.Events) != tt.res.eventLen {
+				t.Errorf("got wrong event len: expected: %v, actual: %v ", tt.res.eventLen, len(agg.Events))
+			}
+			for i := 0; i < tt.res.eventLen; i++ {
+				if !tt.res.wantErr && agg.Events[i].Type != tt.res.eventTypes[i] {
+					t.Errorf("got wrong event type: expected: %v, actual: %v ", tt.res.eventTypes[i], agg.Events[i].Type.String())
+				}
+				if !tt.res.wantErr && agg.Events[i].Data == nil {
+					t.Errorf("should have data in event")
+				}
+			}
+
+			if tt.res.wantErr && !tt.res.errFunc(err) {
+				t.Errorf("got wrong err: %v ", err)
+			}
+		})
+	}
+}
+
+func TestIdpConfigurationRemovedAggregate(t *testing.T) {
+	type args struct {
+		ctx        context.Context
+		existing   *model.Org
+		new        *iam_es_model.IdpConfig
+		provider   *iam_es_model.IdpProvider
+		aggCreator *es_models.AggregateCreator
+	}
+	type res struct {
+		eventLen   int
+		eventTypes []es_models.EventType
+		wantErr    bool
+		errFunc    func(err error) bool
+	}
+	tests := []struct {
+		name string
+		args args
+		res  res
+	}{
+		{
+			name: "remove idp config",
+			args: args{
+				ctx: authz.NewMockContext("orgID", "userID"),
+				existing: &model.Org{
+					ObjectRoot: es_models.ObjectRoot{AggregateID: "AggregateID"},
+					Name:       "Name",
+					IDPs: []*iam_es_model.IdpConfig{
+						{IDPConfigID: "IdpConfigID", Name: "Name"},
+					}},
+				new: &iam_es_model.IdpConfig{
+					ObjectRoot:  es_models.ObjectRoot{AggregateID: "AggregateID"},
+					IDPConfigID: "IdpConfigID",
+					Name:        "Name",
+				},
+				aggCreator: es_models.NewAggregateCreator("Test"),
+			},
+			res: res{
+				eventLen:   1,
+				eventTypes: []es_models.EventType{model.IdpConfigRemoved},
+			},
+		},
+		{
+			name: "remove idp config with provider",
+			args: args{
+				ctx: authz.NewMockContext("orgID", "userID"),
+				existing: &model.Org{
+					ObjectRoot: es_models.ObjectRoot{AggregateID: "AggregateID"},
+					Name:       "Name",
+					IDPs: []*iam_es_model.IdpConfig{
+						{IDPConfigID: "IdpConfigID", Name: "Name"},
+					}},
+				new: &iam_es_model.IdpConfig{
+					ObjectRoot:  es_models.ObjectRoot{AggregateID: "AggregateID"},
+					IDPConfigID: "IdpConfigID",
+					Name:        "Name",
+				},
+				provider: &iam_es_model.IdpProvider{
+					IdpConfigID: "IdpConfigID",
+				},
+				aggCreator: es_models.NewAggregateCreator("Test"),
+			},
+			res: res{
+				eventLen:   2,
+				eventTypes: []es_models.EventType{model.IdpConfigRemoved, model.LoginPolicyIdpProviderCascadeRemoved},
+			},
+		},
+		{
+			name: "existing iam nil",
+			args: args{
+				ctx:        authz.NewMockContext("orgID", "userID"),
+				existing:   nil,
+				aggCreator: es_models.NewAggregateCreator("Test"),
+			},
+			res: res{
+				wantErr: true,
+				errFunc: errors.IsPreconditionFailed,
+			},
+		},
+		{
+			name: "idp config nil",
```

Comment:
remove should be okay if object doesn't exist

Reply:
same as bevore, we have it every where like this, we should not change this here

Reply:
okay